packages/noir-ethereum-api/test/e2e.test.ts

```diff
@@ -28,6 +35,33 @@ describe.concurrent(
       expect(await verifyStorageProof(proofData)).toEqual(true);
     });
 
+    it('anvil smart contract proof verification successes', async () => {
+      const client = createAnvilClient();
+
+      const account: Account = privateKeyToAccount(ANVIL_TEST_ACCOUNT_PRIVATE_KEY);
+
+      const contract: Hex = await client.deployContract({
+        abi: ultraVerifier.abi,
+        account,
+        bytecode: ultraVerifier.bytecode.object as Hex
+      });
+
+      const transaction: TransactionReceipt = await client.waitForTransactionReceipt({ hash: contract });
+      expect(transaction.status).toEqual('success');
+
+      const contractAddress = transaction.contractAddress as Hex;
+
+      const { request: verifyProofRequest } = await client.simulateContract({
+        account,
+        address: contractAddress,
+        abi: ultraVerifier.abi,
+        functionName: 'verify',
+        args: [decodeHexString(proof), Array.from(witnessMap.values())]
+      });
+      const proofVerificationTransactionHash = await client.writeContract(verifyProofRequest);
+      expect(proofVerificationTransactionHash).toSatisfy(isHex);
+    });
+
```

Comment:
I don't think you need to go to simulate flow, Anvil is a simulator, just call the method on the contract:
```ts
const hash = await contract.write.verify([decodeHexString(proof), Array.from(witnessMap.values())])
```
And check transaction receipt status in `except`

[see here](https://viem.sh/docs/contract/getContract)